src/lib/awesomeBarObserver.js

```diff
@@ -0,0 +1,528 @@
+      return false;
+    }
+    this.observedEventsSinceLastFocus.push({
+      awesomeBarState,
+      timestamp: new Date(),
+      type: "onBlur",
+    });
+    await this.afterInteraction();
+    return true;
+  }
+
+  async onKeyDown(keyEvent) {
+    if (await browser.privacyContext.aPrivateBrowserWindowIsOpen()) {
+      // drop the event - do not do any model training
+      return false;
+    }
+    this.observedEventsSinceLastFocus.push({
+      keyEvent,
+      timestamp: new Date(),
+      type: "onKeyDown",
+    });
+    return true;
+  }
+
+  async onKeyPress(keyEvent) {
+    if (await browser.privacyContext.aPrivateBrowserWindowIsOpen()) {
+      // drop the event - do not do any model training
+      return false;
+    }
+    this.observedEventsSinceLastFocus.push({
+      keyEvent,
+      timestamp: new Date(),
+      type: "onKeyPress",
+    });
+    return true;
+  }
+
+  async onInput(awesomeBarState) {
+    if (await browser.privacyContext.aPrivateBrowserWindowIsOpen()) {
+      // drop the event - do not do any model training
+      return false;
+    }
+    this.observedEventsSinceLastFocus.push({
+      awesomeBarState,
+      timestamp: new Date(),
+      type: "onInput",
+    });
+    return true;
+  }
+
+  async onAutocompleteSuggestionsHidden(awesomeBarState) {
+    if (await browser.privacyContext.aPrivateBrowserWindowIsOpen()) {
+      // drop the event - do not do any model training
+      return false;
+    }
+    this.observedEventsSinceLastFocus.push({
+      awesomeBarState,
+      timestamp: new Date(),
+      type: "onAutocompleteSuggestionsHidden",
+    });
+    return true;
+  }
+
+  async onAutocompleteSuggestionsUpdated(awesomeBarState) {
+    if (await browser.privacyContext.aPrivateBrowserWindowIsOpen()) {
+      // drop the event - do not do any model training
+      return false;
+    }
+    this.observedEventsSinceLastFocus.push({
+      awesomeBarState,
+      timestamp: new Date(),
+      type: "onAutocompleteSuggestionsUpdated",
+    });
+    return true;
+  }
+
+  async onAutocompleteSuggestionSelected(awesomeBarState) {
+    if (await browser.privacyContext.aPrivateBrowserWindowIsOpen()) {
+      // drop the event - do not do any model training
+      return false;
+    }
+    this.observedEventsSinceLastFocus.push({
+      awesomeBarState,
+      timestamp: new Date(),
+      type: "onAutocompleteSuggestionSelected",
+    });
+    return true;
+  }
+
+  static isBookmarkOrHistoryStyle(styleString) {
+    const NON_BOOKMARK_OR_HISTORY_STYLES = [
+      "switchtab",
+      "remotetab",
+      "searchengine",
+      "visiturl",
+      "extension",
+      "suggestion",
+      "keyword",
+    ];
+    const styles = new Set(styleString.split(/\s+/));
+    const isNonBookmarkOrHistoryStyle = NON_BOOKMARK_OR_HISTORY_STYLES.some(s =>
+      styles.has(s),
+    );
+    return !isNonBookmarkOrHistoryStyle;
+  }
+
+  /**
+   * @param {string} searchString Self-explanatory hopefully
```

Comment:
The "Self-explanatory hopefully" comment seems redundant - I'd either let `searchString` stand on its own then.

Reply:
Me too. I have a note on updating the eslint configuration to not complain in these cases